prow/plugins/blunderbuss/blunderbuss_test.go

```diff
@@ -230,17 +231,23 @@ var (
 		alternateExpectedRequested []string
 	}{
 		{
-			name:              "one file, 3 leaf reviewers, 1 parent, request 3",
+			name:              "one file, 3 leaf reviewers, 1 parent reviewer, 1 top level reviewer, request 3",
 			filesChanged:      []string{"c.go"},
 			reviewerCount:     3,
 			expectedRequested: []string{"cole", "carl", "chad"},
 		},
 		{
-			name:              "one file, 3 leaf reviewers, 1 parent reviewer, request 4",
+			name:              "one file, 3 leaf reviewers, 1 parent reviewer, 1 top level reviewer, request 4",
 			filesChanged:      []string{"c.go"},
 			reviewerCount:     4,
 			expectedRequested: []string{"cole", "carl", "chad", "charles"},
 		},
+		{
+			name:              "one file, 3 leaf reviewers, 1 parent reviewer, 1 top level reviewer, request 5",
```

Comment:
So this testcase mostly shows that we traverse up? Could we please have a testcase that does not without the associated code changes and does pass with them?

Reply:
This one shows that we take the top level reviewers only when we exhaust the others to match request.
The 2 previous ones show that even though `ben` is alphabetically first, he doesn't go before charles (which is exactly what this PR solves).
I could add another file with charles and ben at the same level to show that within the same layer we still select alphabetically...

Reply:
Okay thanks, that was the explanation I was missing all along and that would have been tremendously helpful in the PR body.

Reply:
Now that I have seen the whole stage 1, stage 2 and stage 3 handling in blunderbuss I am again not convinved that this actualy fixes a bug. The test here can not be adapted to `master`, can we have a test that is adaptable to master (e.G.  create a repo via `localgit`, populate it with a hierarchy of files and owners files , show that we don't fall back incorrectly) and proves that the code  `master` does not handle this properly but the code here does?